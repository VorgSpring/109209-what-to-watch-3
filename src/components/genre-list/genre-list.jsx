import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {GenreItem} from '../genre-item/genre-item.jsx';
import {changeGenre} from '../../actions/filters/genre';
import {
  getGenreTypesSelector
} from '../../selectors/genre/genre';
import {
  getFilterGenreSelector
} from '../../selectors/filters/filters';

export class GenreListComponent extends PureComponent {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(genre) {
    const {onChangeGenre} = this.props;

    onChangeGenre(genre);
  }

  render() {
    const {
      activeGenre,
      genreTypes
    } = this.props;

    return (
      <ul className='catalog__genres-list'>
        {genreTypes.map((genre) => (
          <GenreItem
            key={genre}
            genre={genre}
            isActiveGenre={activeGenre === genre}
            onClick={this.handleClick}
          />
        ))}
      </ul>
    );
  }
}

GenreListComponent.propTypes = {
  genreTypes: PropTypes.arrayOf(
      PropTypes.string
  ).isRequired,
  onChangeGenre: PropTypes.func.isRequired,
  activeGenre: PropTypes.string.isRequired
};

const mapStateToProps = (state) => ({
  genreTypes: getGenreTypesSelector(state),
  activeGenre: getFilterGenreSelector(state)
});

const mapDispatchToProps = (dispatch) => ({
  onChangeGenre: (genre) => dispatch(changeGenre(genre))
});

export const GenreList = connect(
    mapStateToProps, mapDispatchToProps
)(GenreListComponent);
