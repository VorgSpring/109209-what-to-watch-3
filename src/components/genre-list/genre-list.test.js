import React from 'react';
import {GenreListComponent} from './genre-list.jsx';
import renderer from 'react-test-renderer';

describe(`GenreList`, () => {
  it(`should renders correctly`, () => {
    const tree = renderer.create(
        <GenreListComponent
          genreTypes={[`blah`, `blah-blah`]}
          onChangeGenre={() => {}}
          activeGenre={`blah`}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
