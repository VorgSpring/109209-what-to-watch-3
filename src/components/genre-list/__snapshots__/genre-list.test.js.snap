// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreList should renders correctly 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
    >
      blah
    </a>
  </li>
  <li
    className="catalog__genres-item "
    onClick={[Function]}
  >
    <a
      className="catalog__genres-link"
    >
      blah-blah
    </a>
  </li>
</ul>
`;
